test/test_content_type.py

```diff
@@ -0,0 +1,253 @@
+        """
+        route = '/test'
+        app = self.__class__.app_module.app
+        app.routes.pop(route)
+        app._register_handler(handler_type='route',
+                              name='route',
+                              user_handler=handler,
+                              wrapped_handler=handler,
+                              kwargs={'path': route, 'kwargs': {}},
+                              options=None)
+
+    def _shrink_traceback(self, s: str) -> str:
+        """
+        Return a modified version of the given traceback. The first and last
+        lines are kept, and everything inbetween is replaced with a single line
+        of '...'.
+        """
+        if s.startswith('Traceback'):
+            lines = s.split('\n')
+            assert lines[-1] == '', s  # since traceback ends with a '\n'
+            s = '\n'.join([lines[0], '...', lines[-2]])
+        elif s.startswith('<html>'):
+            # Assumes traceback is a json dumped string inside <pre></pre> tags
+            pattern = re.compile(r'(<pre>)'
+                                 r'("Traceback.*?\\n)'  # 1st line of traceback
+                                 r'.*\\n'  # middle lines
+                                 r'(.*)\\n'  # last line of traceback
+                                 r'("</pre>)')
+            s = re.sub(pattern, r'\1\2...\\n\3\4', s)
+        return s
+
+    def _test_route(self,
+                    handler_fn: Callable[[None], Any],
+                    expected_fn: Callable[[bool, bool], Tuple[str, str]]
+                    ):
+        """
+        Verify the response against expected for requests made with various
+        types of `accept` header values.
+        """
+        self._replace_handler(handler_fn)
+        for debug in (False, True):
+            with mock.patch.object(Chalice, 'debug', 1 if debug else 0):
+                for accept, expect_wrapped in [
+                    (None, False),
+                    ('*/*', False),
+                    ('*/*,text/html', False),
+                    ('text/html', True),
+                    ('text/html,*/*', True),
+                    ('*/*;q=0.9,text/html', True),
+                    ('text/html;q=0.9,*/*;q=1.0', False),
+                ]:
+                    with self.subTest(debug=debug, accept=accept):
+                        url = self.base_url.set(path=('test',))
+                        headers = {'accept': accept}
+                        response = requests.get(url, headers=headers)
+                        response_text = self._shrink_traceback(response.text)
+                        expected_text, expected_content_type = expected_fn(debug, expect_wrapped)
+                        self.assertEqual(expected_text, response_text)
+                        self.assertEqual(expected_content_type, response.headers['Content-Type'])
+
+    def test_string(self):
+
+        def route():
+            return '<script />'
+
+        # noinspection PyUnusedLocal
```

Comment:
```diff
Index: test/test_content_type.py
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/test/test_content_type.py b/test/test_content_type.py
--- a/test/test_content_type.py	(revision 145cc0157017310efbc5cbc6dbbfc973970962a5)
+++ b/test/test_content_type.py	(date 1681774413142)
@@ -106,8 +106,7 @@
         def route():
             return '<script />'
 
-        # noinspection PyUnusedLocal
-        def expected(debug: bool, wrapped: bool) -> Tuple[str, str]:
+        def expected(_debug: bool, _wrapped: bool) -> tuple[str, str]:
             text = '<script />'
             content_type = 'application/json'
             return text, content_type
```
https://stackoverflow.com/a/49447358/4171119 is what we usually do in this case.

`tuple` is now generic.